File: modules/common/common-domain/src/main/scala/com/monadial/waygrid/common/domain/model/routing/Route.scala
package com.monadial.waygrid.common.domain.model.routing

import com.monadial.waygrid.common.domain.model.routing.Value.RouteId
import com.monadial.waygrid.common.domain.model.routing.dag.Dag

final case class Route(
  id: RouteId,
  dag: Dag
)

File: modules/admin/ui/src/domain/routing/entities/route.ts
import { Timestamp } from '@/domain/shared'
import { RouteId, createRouteId } from '../value-objects/route-id'
import { DagHash } from '@/domain/traversal'

/**
 * Route - A registered routing configuration
 */
export interface Route {
  readonly id: RouteId
  readonly name: string
  readonly description?: string
  readonly dagHash: DagHash
  readonly isActive: boolean
  readonly createdAt: Timestamp
  readonly updatedAt: Timestamp
}

export function createRoute(params: {
  id: string
  name: string
  description?: string
  dagHash: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}): Route {
  return {
    id: createRouteId(params.id),
    name: params.name,
    description: params.description,
    dagHash: params.dagHash as DagHash,
    isActive: params.isActive,
    createdAt: params.createdAt as Timestamp,
    updatedAt: params.updatedAt as Timestamp,
  }
}
